[WIP] Adds include extractor to the processing pipeline

HEskandari opened the pull request:
This adds the 'include' extraction to the processing pipeline. There's a CSharp version that's picked based on detected language and there's a no-op one which it falls back to.

I still need to figure out how to expose that to the end-user to provide their custom implementation for other languages.

SimonCropp commented on CaptureSnippets/IncludeExtracotrs/CSharpUsingExtractor.cs:
```diff
@@ -0,0 +1,27 @@
+{
+    public class CSharpUsingExtractor : IIncludeExtractor
+    {
+        const string UsingPattern = @"(?:using\s)(?<ns>.*)(?:;)";
```
i would much prefer you use normal string manipulation instead of a regex. it will be faster, use less mem and less likely to have bugs

SimonCropp commented on CaptureSnippets/IncludeExtracotrs/IIncludeExtractor.cs:
```diff
@@ -0,0 +1,7 @@
+namespace CaptureSnippets.IncludeExtracotrs
```
typo in dir Extracotrs

SimonCropp commented on CaptureSnippets/IncludeExtracotrs/IIncludeExtractor.cs:
```diff
@@ -0,0 +1,7 @@
+namespace CaptureSnippets.IncludeExtracotrs
+{
+    public interface IIncludeExtractor
```
i would prefer we pass in a delegate instead of introduce a new interface

HEskandari commented:
@SimonCropp Please review. In summary:

- Removed the interface and used a delegate
- Removed the regex
- Now handles duplicate cs namespaces, simple comments and aliases
- Added more tests
